src/node_api.cc

```diff
@@ -570,8 +570,7 @@ class SetterCallbackWrapper
 
   /*virtual*/
   void SetReturnValue(napi_value value) override {
-    node::FatalError("napi_set_return_value",
-      "Cannot return a value from a setter callback.");
+    assert(false);  // Cannot return a value from a setter callback
```

Comment:
Yes, I was aware `assert()` is a no-op in a release build. I don't think we need a fatal error here. If someone tries to set a return value from a property setter, it's not so bad to just ignore it in a release build.